src/esp/metadata/attributes/DatasetAttributes.h

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_METADATA_ATTRIBUTES_DATASETATTRIBUTES_H_
+#define ESP_METADATA_ATTRIBUTES_DATASETATTRIBUTES_H_
+
+/** @file
+ * @brief Class @ref esp::metadata::DatasetMetadata.  This class will hold
+ * relevant data and configurations for a specific dataset.
```

Comment:
What is a dataset?

Reply:
A dataset is a user-defined collection of stages, objects, lighting configurations, scene instances, navmeshes and semantic scene descriptor meshes.  

The DatasetAttributes structure provides a mechanism for organizing the metadata describing these constructs into a kind of library.

Reply:
This has some overloading with Habitat-Lab datasets...

Reply:
Renamed to scene_dataset_config and SceneDatasetAttributes to reflect discussions on the above point.